File: SpringMVC_API/src/main/java/paul/spring/mvc/practice/tasks/springdata/TaskServiceConfig.java
package paul.spring.mvc.practice.tasks.springdata;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import paul.spring.mvc.practice.tasks.TaskService;

@Configuration
public class TaskServiceConfig {
    @Bean
    TaskService taskService(final TaskSpringDataRepository taskSpringDataRepository) {
        return new TaskSpringDataService(taskSpringDataRepository);
    }
}

File: SpringWebflux_Kotlin/todos/src/main/kotlin/com/paul/kotlin/todos/tasks/springdata/TaskServiceConfig.kt
package com.paul.kotlin.todos.tasks.springdata

import com.paul.kotlin.todos.tasks.TaskService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class TaskServiceConfig {

    @Bean
    @ConditionalOnMissingBean
    fun taskService(taskSpringDataRepository: TaskSpringDataRepository): TaskService {
        return TaskSpringDataService(taskSpringDataRepository);
    }
}
